plenum/server/view_change/view_changer.py

```diff
@@ -0,0 +1,626 @@
+        # sent otherwise no.
+        canSendInsChange, cooldown = self.insChngThrottler.acquire()
+        if canSendInsChange:
+            logger.info(
+                "{}{} sending an instance change with view_no {}"
+                " since {}".format(
+                    VIEW_CHANGE_PREFIX,
+                    self,
+                    view_no,
+                    suspicion.reason))
+            logger.info("{}{} metrics for monitor: {}"
+                        .format(MONITORING_PREFIX, self,
+                                self.node.monitor.prettymetrics))
+            msg = self._create_instance_change_msg(view_no, suspicion.code)
+            self.send(msg)
+            # record instance change vote for self and try to change the view
+            # if quorum is reached
+            self._on_verified_instance_change_msg(msg, self.name)
+        else:
+            logger.debug(
+                "{} cannot send instance change sooner then {} seconds".format(
+                    self, cooldown))
+
+    # noinspection PyAttributeOutsideInit
+    def initInsChngThrottling(self):
+        windowSize = self.node.config.ViewChangeWindowSize
+        ratchet = Ratchet(a=2, b=0.05, c=1, base=2, peak=windowSize)
+        self.insChngThrottler = Throttler(windowSize, ratchet.get)
+
+    def _create_instance_change_msg(self, view_no, suspicion_code):
+        return InstanceChange(view_no, suspicion_code)
+
+    def _on_verified_instance_change_msg(self, msg, frm):
+        view_no = msg.viewNo
+
+        if not self.instanceChanges.hasInstChngFrom(view_no, frm):
+            self.instanceChanges.addVote(msg, frm)
+            if view_no > self.view_no:
+                self.do_view_change_if_possible(view_no)
+
+    def do_view_change_if_possible(self, view_no):
+        # TODO: Need to handle skewed distributions which can arise due to
+        # malicious nodes sending messages early on
+        can, whyNot = self.canViewChange(view_no)
+        if can:
+            logger.info("{}{} initiating a view change to {} from {}".
+                        format(VIEW_CHANGE_PREFIX, self, view_no, self.view_no))
+            self.propagate_primary = False
+            self.startViewChange(view_no)
+        else:
+            logger.debug(whyNot)
+        return can
+
+    def _start_view_change_if_possible(self, view_no) -> bool:
+        ind_count = len(self._next_view_indications[view_no])
+        if self.quorums.propagate_primary.is_reached(ind_count):
+            logger.info(
+                '{}{} starting view change for {} after {} view change '
+                'indications from other nodes'.format(
+                    VIEW_CHANGE_PREFIX, self, view_no, ind_count))
+            self.propagate_primary = True
+            self.startViewChange(view_no)
+            return True
+        return False
+
+    def canViewChange(self, proposedViewNo: int) -> (bool, str):
```

Comment:
I agree. Will rename that method. For other ones I suggest to do the work in scope of further refactoring tasks that are coming because the list of methods could be changed  significantly.